feat: Add support for custom resources

stepanstipl opened the pull request:
~*WIP: Still needs:*~
- [x] ~docs update~
- [x] ~(at least) some test coverage~
- [x] ~test to test the custom rule~
- [x] ~squash before merge~

---

This PR adds support for detecting and flagging custom resources.
```
kubent -a "ManagedCertificate.v1.networking.gke.io"
```

The `-a` flag, or it's full form `--additional-kind`, can be used multiple times.
The expected format is full `Kind.version.group.com` form - e.g. `ManagedCertificate.v1.networking.gke.io`.


Closes #44

david-doit-intl commented:
Could we add a test where additional is not  []string{}?

david-doit-intl commented on pkg/judge/rego.go:
```diff
@@ -19,17 +24,24 @@ type RegoJudge struct {
 
-	fs.Walk(statikFS, "/",
+	err = fs.Walk(statikFS, "/",
 		func(path string, info os.FileInfo, err error) error {
```
this is too large now, please make it a separate go function

stepanstipl replied:
This should be addressed now.

david-doit-intl reviewed (changes requested):
Some minor cleanup, but looks good

stepanstipl commented:
I think this is ready for another review. I did rebase and refactor to accommodate the changes introduced with the use of the embed FS, addressed the too long function mentioned above and added unit tests.

What still needs to be done before merge:
- The docs still need to be updated
- and I plan to add at least one "functional" test to cover the checking for additional resource

stepanstipl commented:
@david-doit-intl sorry, I can't see where you commented as I noticed the failing test and was fixing the fmt error. If it was that, it's fixed, if smth. else, can you please comment again? :)

david-doit-intl commented on pkg/rules/rules_test.go:
```diff
@@ -26,3 +29,93 @@ func TestFetchRules(t *testing.T) {
+
+func TestFetchRulesWithAdditionalResources(t *testing.T) {
+	var expected []string
+	root := "rego/"
```
use the const

stepanstipl replied:
Done.

david-doit-intl commented on pkg/rules/rules_test.go:
```diff
@@ -26,3 +29,93 @@ func TestFetchRules(t *testing.T) {
+	var expected []string
+	root := "rego/"
+	err := filepath.Walk(root, func(path string, info os.FileInfo, err error) error {
+		if info.Name() != "rego" {
```
here too

stepanstipl replied:
Done.

david-doit-intl commented on cmd/kubent/main.go:
```diff
@@ -79,13 +80,14 @@ func initCollectors(config *config.Config) []collector.Collector {
 
-	log.Logger = log.Output(zerolog.ConsoleWriter{Out: os.Stderr})
-	zerolog.SetGlobalLevel(zerolog.InfoLevel)
 	if config.Debug {
```
might want to move that too!

stepanstipl commented:
Added the "integration" test to test the actual custom rule functionality. It's in a separate package `test`, as it's using other packages and can't sit under rules (cause cyclic imports). I think it can serve as a starting point for implementing tests for the individual rules.

I think it's ready for final review @david-doit-intl
